spring-web/src/main/java/org/springframework/web/client/RestTemplate.java

```diff
@@ -684,8 +685,16 @@ public void delete(URI url) throws RestClientException {
 			response = request.execute();
 			handleResponse(url, method, response);
 			if (responseExtractor != null) {
-				return responseExtractor.extractData(response);
-			}
+                T entityData = responseExtractor.extractData(response);
+                if(entityData instanceof ResponseEntity){
```

Comment:
- [ ] Why check `ResponseEntity` only? I think should be check the `InputStreamSource` too.